libs/sdk-core/src/swap_in/notification.rs

```diff
@@ -0,0 +1,33 @@
+use reqwest::Body;
+use serde_json::json;
+use tonic::codegen::http::header::CONTENT_TYPE;
+
+use crate::error::{SdkError, SdkResult};
+use crate::input_parser::get_reqwest_client;
+
+/// Registers for a swap tx notification. When a new transaction to the specified `swap_address`
+/// is confirmed, a callback will be triggered to the specified `webhook_url`.
+pub(crate) async fn register_swap_tx_webhook(
```

Comment:
Can we make this function only accessible to the swapper?

Reply:
Done: 91e169eaed1e727d4e7e5796ee9fa978b719fce4